getwork.go

```diff
@@ -1,298 +0,0 @@
-		}
-	}
-
-	// Create and return the new HTTP client potentially configured with a
-	// proxy and TLS.
-	client := http.Client{
-		Transport: &http.Transport{
-			Dial:            dial,
-			TLSClientConfig: tlsConfig,
-		},
-	}
-	return &client, nil
-}
-
-type getWorkResponseJson struct {
-	Result struct {
-		Data   string
-		Target string
-	}
-	Error *struct {
-		Code    int
-		Message string
-	}
-}
-
-type getWorkSubmitResponseJson struct {
-	Result bool
-	Error  *struct {
-		Code    int
-		Message string
-	}
-}
-
-const (
-	MaxIdleConnections int = 20
-	RequestTimeout     int = 5
-)
-
-// GetWork makes a getwork RPC call and returns the result (data and target)
-func GetWork() (*work.Work, error) {
-	// Generate a request to the configured RPC server.
-	protocol := "http"
-	if !cfg.NoTLS {
-		protocol = "https"
-	}
-	url := protocol + "://" + cfg.RPCServer
-	jsonStr := []byte(`{"jsonrpc": "2.0", "method": "getwork", "params": [], "id": 1}`)
-	bodyBuff := bytes.NewBuffer(jsonStr)
-	httpRequest, err := http.NewRequest("POST", url, bodyBuff)
-	if err != nil {
-		return nil, err
-	}
-	httpRequest.Close = true
-	httpRequest.Header.Set("Content-Type", "application/json")
-
-	// Configure basic access authorization.
-	httpRequest.SetBasicAuth(cfg.RPCUser, cfg.RPCPassword)
-
-	// Create the new HTTP client that is configured according to the user-
-	// specified options and submit the request.
-	httpClient, err := newHTTPClient(cfg)
-	if err != nil {
-		return nil, err
-	}
-	httpResponse, err := httpClient.Do(httpRequest)
-	if err != nil {
-		return nil, err
-	}
-
-	body, err := ioutil.ReadAll(httpResponse.Body)
-	httpResponse.Body.Close()
-	if err != nil {
-		err = fmt.Errorf("error reading json reply: %v", err)
-		return nil, err
-	}
-
-	if httpResponse.Status != "200 OK" {
-		return nil, fmt.Errorf("HTTP %s: %s", httpResponse.Status, body)
-	}
-
-	var res getWorkResponseJson
-	err = json.Unmarshal(body, &res)
-	if err != nil {
-		return nil, err
-	}
-
-	if res.Error != nil {
-		return nil, fmt.Errorf("JSONRPC Error %d: %s", res.Error.Code,
-			res.Error.Message)
-	}
-
-	data, err := hex.DecodeString(res.Result.Data)
-	if err != nil {
-		return nil, err
-	}
-	if len(data) != 192 {
-		return nil, fmt.Errorf("Wrong data length: got %d, expected 192",
-			len(data))
-	}
-	target, err := hex.DecodeString(res.Result.Target)
-	if err != nil {
-		return nil, err
-	}
-	if len(target) != 32 {
-		return nil, fmt.Errorf("Wrong target length: got %d, expected 32",
-			len(target))
-	}
-
-	bigTarget := new(big.Int)
-	bigTarget.SetBytes(util.Reverse(target))
-
-	var workData [192]byte
-	copy(workData[:], data)
-	givenTs := binary.LittleEndian.Uint32(
-		workData[128+4*work.TimestampWord : 132+4*work.TimestampWord])
-	w := work.NewWork(workData, bigTarget, givenTs, uint32(time.Now().Unix()), true)
-
-	w.Target = bigTarget
-
-	return w, nil
-}
-
-// GetPoolWork gets work from a stratum enabled pool
-func GetPoolWork(pool *stratum.Stratum) (*work.Work, error) {
```

Comment:
I think removing this is going to a problem for supporting dcrpool (really stratum in general).